client/me/purchases/marketplace-subscriptions-dialog/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button, Dialog } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { Fragment, ReactChild, useState } from 'react';
+import FormSectionHeading from 'calypso/components/forms/form-section-heading';
+
+interface ButtonProps {
+	action: string;
+	label: ReactChild;
+	onClick: () => void;
+	isPrimary?: boolean;
+	busy?: boolean;
+}
+
+interface MarketPlaceSubscriptionsDialogProps {
+	planName: string;
+	closeDialog: () => void;
+	removePlan: () => void;
+	isDialogVisible: boolean;
+	isRemoving: boolean;
+	activeSubscriptions: Array< { id: number; productName: string } >;
+	removeAllSubscriptions: () => void;
+}
+
+export const MarketPlaceSubscriptionsDialog = ( {
+	planName,
+	closeDialog,
+	removePlan,
+	isDialogVisible,
+	isRemoving,
+	activeSubscriptions,
+	removeAllSubscriptions,
+}: MarketPlaceSubscriptionsDialogProps ): JSX.Element => {
+	const translate = useTranslate();
+
+	const [ removeAction, setRemoveAction ] = useState( '' );
+
+	const buttons: ButtonProps[] = [
+		{
+			action: 'cancel',
+			label: translate( 'Cancel' ),
+			onClick: closeDialog,
+		},
+		{
+			action: 'removePlan',
+			label: translate( 'Remove Plan' ),
+			onClick: removePlan,
+		},
+		{
+			isPrimary: true,
+			action: 'removeAllSubscriptions',
+			label: translate( 'Remove All Subscriptions' ),
+			onClick: removeAllSubscriptions,
+			busy: true,
+		},
+	];
+
+	const renderButton = ( { action, label, onClick, isPrimary }: ButtonProps ) => (
+		<Button
+			disabled={ isRemoving && removeAction !== action }
+			busy={ isRemoving && removeAction === action }
+			onClick={ () => {
+				onClick();
+				setRemoveAction( action );
+			} }
+			primary={ isPrimary }
+			data-e2e-button={ action }
+			data-tip-target={ `dialog-base-action-${ action }` }
+		>
+			{ label }
+		</Button>
+	);
+
+	return (
+		<Dialog
+			buttons={ buttons.map( ( b ) => renderButton( b ) ) }
+			className="marketplace-subscriptions-dialog"
+			isVisible={ isDialogVisible }
+			onClose={ closeDialog }
+		>
+			<Fragment>
+				<FormSectionHeading>
```

Comment:
There's a type error here, which is why the unit tests are failing:

```
81:  Property 'className' is missing in type '{ children: ReactChild; }' but required in type '{ [x: string]: any; className: any; children: any; }'.
```

The problem is that the definition of `FormSectionHeading` is in JS and TS is assuming that all its props are required. You can resolve this either by adding JSDoc to that function or setting a default prop for `className` like this:

```js
const FormSectionHeading = ( { className = '', children, ...otherProps } ) => (
```

Reply:
Great tips, thanks @sirbrillig! 🎉 I've added JSDoc annotations, now the tests pass ✅